cmd/root.go

```diff
@@ -116,101 +119,108 @@ func initConfig() {
-	viper.BindPFlag("tlskey", rootCmd.Flags().Lookup("tlskey"))
+	_ = viper.BindPFlag("tlskey", rootCmd.Flags().Lookup("tlskey"))
 
 	// tlscacert
-	rootCmd.Flags().String("tlscacert", "", "Client certificate CA on NATS connecctions.")
-	viper.BindPFlag("tlscacert", rootCmd.Flags().Lookup("tlscacert"))
+	rootCmd.Flags().String("tlscacert", "", "Client certificate CA on NATS connections.")
+	_ = viper.BindPFlag("tlscacert", rootCmd.Flags().Lookup("tlscacert"))
 
 	// http-tlscert
 	rootCmd.Flags().String("http-tlscert", "", "Server certificate file (Enables HTTPS).")
-	viper.BindPFlag("http-tlscert", rootCmd.Flags().Lookup("http-tlscert"))
+	_ = viper.BindPFlag("http-tlscert", rootCmd.Flags().Lookup("http-tlscert"))
 
 	// http-tlskey
 	rootCmd.Flags().String("http-tlskey", "", "Private key for server certificate (used with HTTPS).")
-	viper.BindPFlag("http-tlskey", rootCmd.Flags().Lookup("http-tlskey"))
+	_ = viper.BindPFlag("http-tlskey", rootCmd.Flags().Lookup("http-tlskey"))
 
 	// http-tlscacert
 	rootCmd.Flags().String("http-tlscacert", "", "Client certificate CA for verification (used with HTTPS).")
-	viper.BindPFlag("http-tlscacert", rootCmd.Flags().Lookup("http-tlscacert"))
+	_ = viper.BindPFlag("http-tlscacert", rootCmd.Flags().Lookup("http-tlscacert"))
 
 	// http-user
 	rootCmd.Flags().String("http-user", "", "Enable basic auth and set user name for HTTP scrapes.")
-	viper.BindPFlag("http-user", rootCmd.Flags().Lookup("http-user"))
+	_ = viper.BindPFlag("http-user", rootCmd.Flags().Lookup("http-user"))
 
 	// http-pass
 	rootCmd.Flags().String("http-pass", "", "Set the password for HTTP scrapes. NATS bcrypt supported.")
-	viper.BindPFlag("http-pass", rootCmd.Flags().Lookup("http-pass"))
+	_ = viper.BindPFlag("http-pass", rootCmd.Flags().Lookup("http-pass"))
 
 	// prefix
 	rootCmd.Flags().String("prefix", "", "Replace the default prefix for all the metrics.")
-	viper.BindPFlag("prefix", rootCmd.Flags().Lookup("prefix"))
+	_ = viper.BindPFlag("prefix", rootCmd.Flags().Lookup("prefix"))
 
 	// observe
 	rootCmd.Flags().String("observe", "", "Listen for observation statistics based on config files in a directory.")
-	viper.BindPFlag("observe", rootCmd.Flags().Lookup("observe"))
+	_ = viper.BindPFlag("observe", rootCmd.Flags().Lookup("observe"))
 
 	// jetstream
 	rootCmd.Flags().String("jetstream", "", "Listen for JetStream Advisories based on config files in a directory.")
-	viper.BindPFlag("jetstream", rootCmd.Flags().Lookup("jetstream"))
+	_ = viper.BindPFlag("jetstream", rootCmd.Flags().Lookup("jetstream"))
 
 	// accounts
 	rootCmd.Flags().Bool("accounts", false, "Export per account metrics")
-	viper.BindPFlag("accounts", rootCmd.Flags().Lookup("accounts"))
+	_ = viper.BindPFlag("accounts", rootCmd.Flags().Lookup("accounts"))
+
+	// log-level
+	rootCmd.Flags().String("log-level", "", "Log level, one of: trace|debug|info|warn|error|fatal|panic (default: info)")
```

Comment:
This should probably be added in the legacyMap like --http_tlscacert so that it can be read after a --

Reply:
It is not a legacy flag though - it should not exist anywhere already, so we don't need to support the non-POSIX style for it

Reply:
wasn't the legacy map where we were also changing the - to a _ though? All the other flags have a replacement for that as well.

Reply:
The Legacy map was to take all of the flags from `v0.22` and ensure that they converted to POSIX flags, then throw a Deprecation warning if a Legacy tag is detected.  New flags introduced after `v0.22` are not Legacy flags, and should only be accepted as POSIX flags

Reply:
Right, but there are replacements for each one that has a _ to convert it to a - even with a -- prefix. I was told this was because we wanted to accept posix flags with a _ and convert it to the corresponding posix flag with a - just like here: https://github.com/nats-io/nats-surveyor/blob/c4e3afe68bbfa9ebd60d84cbe8865ab1c55366d5/cmd/root.go#L60

Reply:
oh that is because we changed the convention for flags with a separator from `_` to `-` so that flag actually changed...  I think we just want the new flags with a separator to be `-` only